src/duke/storage/TaskStorage.java

```diff
@@ -0,0 +1,113 @@
+package duke.storage;
+
+import duke.exception.UnknownTaskTypeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * TaskStorage saves and loads task data.
+ */
+public class TaskStorage {
+
+    private Path folderPath;
+    private File folderFile;
+    private Path taskFilePath;
+    private File taskFile;
+    private Path isDoneFilePath;
+    private File isDoneFile;
```

Comment:
Path/File names might sound like boolean variables here

Reply:
Yeah, the naming is not very good, the file is used to store information of if each saved task is done, so I named it as isDoneFile. If you have any better naming suggestions please feel free to tell me.